### PR TITLE
Extractscode

### PR DESCRIPTION
**Pull request by ovinis**
Merging in code for extract commands 

- createextracts
- decryptextracts
- encryptextracts
- reencryptextracts
- refreshextracts

### CODE REVIEW

File: pythontabcmd2/commands/extracts/encrypt_extracts_command.py [hunk 1]
```diff
@@ -23,4 +31,10 @@ def run_command(self):
         self.encrypt_extract(server_object)
 
     def encrypt_extract(self, server):
-        pass
+        try:
+            site_id = SiteCommand.find_site_id(server, self.site_name)
+            job = server.sites.encrypt_extracts(site_id)
+            self.logger.info("Extract encrypted Successfully with "
```
Comment (jacalata) [inline pythontabcmd2/commands/extracts/encrypt_extracts_command.py:37]: I think this command just starts encryption, we don't know if it finished/succeeded? Same for anything that returns a job id.

File: pythontabcmd2/commands/extracts/extracts_command.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+class ExtractsCommand(Commands):
+    """
+    Base class for extracts group of commands
+    """
+    def __init__(self, args):
+        super().__init__(args)
+
+    @staticmethod
```
Comment (jacalata) [inline pythontabcmd2/commands/extracts/extracts_command.py:11]: these methods seem useful for multiple commands, like delete and publish. Do we have a general place they can go?

File: pythontabcmd2/parsers/create_extracts_parser.py [hunk 3]
```diff
@@ -29,15 +29,16 @@ def create_extracts_parser():
...
                                            help='Create encrypted extract')
         create_extract_parser.add_argument('--include-all',
+                                           action='store_false',
                                            help='Include all embedded data '
                                                 'sources within target'
                                                 ' workbook. Only available '
                                                 'when creating extracts '
                                                 'for workbook.')
-        create_extract_parser.add_argument('--project',
+        create_extract_parser.add_argument('--project', default='default',
```
Comment (jacalata) [inline pythontabcmd2/parsers/create_extracts_parser.py:41]: shouldn't the default for a project be None?

**Review by jacalata — APPROVED** [review]

File: pythontabcmd2/parsers/create_extracts_parser.py (hunk 3 above)
Comment (ovinis) [reply pythontabcmd2/parsers/create_extracts_parser.py]: Thanks for catching it. It should be None

File: pythontabcmd2/commands/extracts/extracts_command.py (hunk 2 above)
Comment (ovinis) [reply pythontabcmd2/commands/extracts/extracts_command.py]: Yes, I moved it to the commands.py file that has the `class Commands`. All commands inherit this class so it seems like a good place.

File: pythontabcmd2/commands/extracts/encrypt_extracts_command.py (hunk 1 above)
Comment (ovinis) [reply pythontabcmd2/commands/extracts/encrypt_extracts_command.py]: Fixed it, Thank you!

**Pull request merged**